### ISSUE TITLE
Export-Excel, CellStyleSB: $LastColumn contains a wrong value

### ISSUE THREAD

**Issue by roberto-cantu** [dfinke/ImportExcel#960]
Seen in v7.1.1

In Export-Excel.ps1, before executing the scriptblock CellStyleSB, the calculation of the $LastColumn returns a wrong value.
Example. If the worksheet is A1:R536, $LastColumn is computed as "R53".
This because the RegExp in the statement below uses the `\w` character class, which include digits, and the * "eats" as many digits as it can but the very last:

            $LastColumn = $ws.Dimension.Address -replace "^.*:(\w*)\d+$" , '$1'

This could be fixed just by replacing the `\w*` with `\D+`.

Thank you

**Comment by dfinke** [comment]
Thanks @roberto-cantu, not sure if you mis-typed your solution?

Not sure which one I'll move forward with.

```powershell
$address = 'A1:R536'

$address -replace "^.*:(\w*)\d+$" , '$1' # original
$address -replace "^.*:(\D+)\d+$" , '$1' # suggested
$address -replace "^.*:(\w*)\D+$" , '$1' # revised - works

# may go with this
$pos = $address.IndexOfAny(':')
$address.Substring($pos + 1)
```

**Comment by roberto-cantu** [comment]
Hello Doug,

I've tried your statements, the last one does not work.
See them at work on my Win10

PS C:\> $address = 'A1:R536'

PS C:\>
PS C:\> $address -replace "^.*:(\w*)\d+$" , '$1' # original
R53
PS C:\> $address -replace "^.*:(\D+)\d+$" , '$1' # suggested
R
PS C:\> $address -replace "^.*:(\w*)\D+$" , '$1' # DOES NOT WORK
A1:R536
PS C:\> get-host


Name             : ConsoleHost
Version          : 5.1.18362.1171
InstanceId       : cb880b5e-bac4-45c4-bde8-2c2100c74f0b
UI               :
System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : it-IT
CurrentUICulture : it-IT
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace

By the way the last regexp is not expected to work.
\D means "not \d", so it matches a non-digit
\w is the shortcut for [A-Za-z0-9_] so it matches digits too, not only
letters

Because in $address after the ":" we can have 1+ letters followed by 1+
digits, the (\D+) should be fine here.
Doesn't it work for you ?
Thank you

Il giorno gio 26 nov 2020 alle ore 16:41 Doug Finke <
notifications@github.com> ha scritto:

**Comment by scriptingstudio** [comment]
`"^.*:([\D\W]+)\d+$"` will work always

**Comment by dfinke** [comment]
I'll be dropping the RegEx. I always go by the saying, I had a problem, then I used a RegEx to solve it, then I had two problems.
The `SubString()` works fine and is more readable.

**Comment by roberto-cantu** [comment]
My tests

PS C:\> $address
A1:R536
PS C:\> $pos = $address.IndexOfAny(':')
PS C:\> $address.Substring($pos + 1)

R536
PS C:\> $address = 'A1:CD234'

PS C:\> $pos = $address.IndexOfAny(':')

PS C:\> $address.Substring($pos + 1)

CD234
PS C:\>

So the $LastColumn would keep the row number.
Is this intended ?

Il giorno gio 26 nov 2020 alle ore 21:36 Doug Finke <
notifications@github.com> ha scritto:

**Comment by dfinke** [comment]
To be determined. That code is older, implemented ~4yrs ago it was an early experiment. Unfortunately there are no unit tests for it.

You can always use `Set-CellStyle` directly rather than using this parameter.

Needs to be reviewed. Regardless, returning `R53` and not `R536` seems wrong. 

We'll see.

**Comment by roberto-cantu** [comment]
Hello Doug,

until a few days ago, I had Import-Excel v5.0.1 installed, and in that
version $LastColumn's value was 'R'.
So I took for granted that also the latest version had to do the same, and
that's why I suggested that statement to you, which returns only 'R'.

Now I understand that the latest version is returning 'R536' instead of
'R', which is the bottom-right cell ID.


Il giorno ven 27 nov 2020 alle ore 18:31 Doug Finke <
notifications@github.com> ha scritto:
